File: core/amber/src/main/scala/edu/uci/ics/texera/web/auth/UserRoleAuthorizer.java
package edu.uci.ics.texera.web.auth;

import io.dropwizard.auth.Authorizer;

public class UserRoleAuthorizer implements Authorizer<SessionUser> {

	@Override
	public boolean authorize(SessionUser user, String role) {

		// currently we do not have multiple roles, grant any SessionUser.
		return true;
	}
}

File: core/file-service/src/main/scala/edu/uci/ics/texera/service/auth/UserRoleAuthorizer.scala
package edu.uci.ics.texera.service.auth

import edu.uci.ics.texera.dao.jooq.generated.enums.UserRole
import io.dropwizard.auth.Authorizer
import jakarta.ws.rs.container.ContainerRequestContext

object UserRoleAuthorizer extends Authorizer[SessionUser] {
  override def authorize(
      user: SessionUser,
      role: String,
      requestContext: ContainerRequestContext
  ): Boolean = {
    user.isRoleOf(UserRole.valueOf(role))
  }
}
